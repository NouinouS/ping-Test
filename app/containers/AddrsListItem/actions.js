/*
 *
 * AddrsListItem actions
 *
 */

import { ADD_ADDRS } from './constants';

export function addAddrs(addrs) {
  return {
    type: ADD_ADDRS,
    addrs
  };
}
