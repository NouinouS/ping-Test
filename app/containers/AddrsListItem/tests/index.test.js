/**
 *
 * Tests for AddrsListItem
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { AddrsListItem, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { browserHistory } from 'react-router-dom';
import { addAddrs } from '../actions';

const renderComponent = (store, props = {}) =>
  renderer.create(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddrsListItem {...props} />
        </IntlProvider>
      </Provider>
    )
    .toJSON();

describe('<AddrsListItem />', () => {
  let item;
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  beforeEach(() => {
    item = {
      id: 1,
      addrs: "192.168.1.1"
    };
  });

  it('should render a ListItem', () => {
    const renderedComponent = renderComponent(store, { item });
    expect(renderedComponent).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onAddAddrs', () => {

      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onAddAddrs).toBeDefined();
      });

      it('should dispatch addAddrs when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch, { item: {addrs: "192.168.1.1"}});
        result.onAddAddrs({ addrs: "192.168.1.1" });
        expect(dispatch).toHaveBeenCalledWith(addAddrs("192.168.1.1"));
      });

    });

  });
});
