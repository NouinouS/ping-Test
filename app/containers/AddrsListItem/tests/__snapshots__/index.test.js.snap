// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddrsListItem /> should render a ListItem 1`] = `
<div>
  <button>
    Add
  </button>
  192.168.1.1
</div>
`;
